TEST_APPS/testcases/nfc/test_nfc.py

```diff
@@ -0,0 +1,288 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+
+from icetea_lib.bench import Bench
+from nfc_clf_wrapper import ContactlessCommandRunner
+import nfc_messages
+import time
+from mbed_clitest.tools.tools import test_case
+import icetea_lib.tools.asserts as asserts
+from nfc_messages import NfcErrors
+from nfc_cli_helper import CliHelper
+
+
+class CreamSconeTests(Bench, CliHelper):
+    """
+    This test wrapper requires a usb connected contactless card reader dongle, and allows E2E testing.
+    See readme file for details
+    """
+    def __init__(self, **kwargs):
+        testcase_args = {
+                       'title':"NFC tests with a reader",
+                       'status':"development",
+                       'purpose':"NFC e2e",
+                       'component':["NFC"],
+                       'type':"regression",
+                       'requirements':{
+                           "duts": {
+                               '*': {
+                                   "count": 1,
+                                   "type": "hardware",
+                                   "application": {
+                                       "name": "TEST_APPS-device-nfcapp"
+                                   }
+                               },
+                               "1": {"nick": "dev1"}
+                           }
+                       }
+        }
+        testcase_args.update(kwargs)
+        Bench.__init__(self, **testcase_args)
+
+    def setup(self):
+        try:
+            self.clf = ContactlessCommandRunner()
+            self.clf.parse("mute")
+        except:
+            raise asserts.TestStepFail("Could not find NFC reader")
+
+    def teardown(self):
+        self.logger.info("Test teardown: Reboot target...")
+        self.reset_dut()
+
+
+
+@test_case(CreamSconeTests)
+def test_nfce2e_target_found(self):
+    """
+    smoke - Our emulated tag is detectable
+    """
+
+    response = self.nfc_command("dev1", "iseeprom")
+    eeprom = response.parsed['iseeprom']
+    self.nfc_command("dev1", "initnfc")
+    if not eeprom:
+        self.nfc_command("dev1", "start")
+
+    self.clf.parse("connect")
+    tag = self.clf.clf_response()
+    asserts.assertNotNone(tag, "Could not connect to any tag")
+
+
+@test_case(CreamSconeTests)
+def test_nfce2e_type4_found(self):
+    """
+    check - Type 4 tag is detected contactless
+    """
+
+    response = self.nfc_command("dev1", "iseeprom")
+    eeprom = response.parsed['iseeprom']
+    self.nfc_command("dev1", "initnfc")
+    if not eeprom:
+        self.nfc_command("dev1", "start")
+
+    self.clf.parse("connect")
+    tag = self.clf.clf_response()
+    asserts.assertNotNone(tag, "Could not connect to any tag")
+
+    asserts.assertEqual(tag.type, 'Type4Tag', "Tag of type Type4Tag not found")
+
+
+@test_case(CreamSconeTests)
+def test_nfce2e_smartposter(self):
```

Comment:
Where are the opposite tests (writing using external initiator, reading back using Mbed OS API)?